File: java/Lezione34_polimorfismo.java
public class Lezione34_polimorfismo {
    public static void main(String[] args) {
        
        // Creazione degli oggetti specifici
        Studente studente1 = new Studente("Luca", "Rossi", 24, "Matematica");
        Studente studente2 = new Studente("Orazio", "Esposito", 22, "Storia");
        Insegnante insegnante1 = new Insegnante("Marco", "Verdi", 45, "Storia");
        
        // POLIMORFISMO: Creiamo un array di 'Persona1' che ospita sottoclassi diverse.
        // Anche se sono oggetti diversi, Java li vede tutti come "Persone".
        Informazione[] classe = {studente1, studente2, insegnante1};

        System.out.println("--- INIZIO APPELLO IN CLASSE ---\n");

        // Ciclo for-each: per ogni 'persona' dentro l'array 'classe'
        for(Informazione persona : classe) {
            // Qui avviene la magia: Java capisce da solo se chiamare il saluto 
            // dello studente o dell'insegnante in base all'oggetto REALE.
            persona.saluta();
        }

        System.out.println("\n--- FINE APPELLO ---");
    }
}

File: C#/Lezione34_polimorfismo.cs
using System;
using PersonaPencipale;
using System.Collections.Generic;

namespace Lezione34 {
    public class Lezione34_polimorfismo {
        static void Main(string[] args) {
            var persone = new List<Persona>{
              new Studente("Luca", "Rossi", 24, "4A"),
              new Insegnante("Anna", "neri", 45, "Storia")  
            };

            // Grazie al POLIMORFISMO, C# capisce a runtime il vero tipo dell'oggetto!
            foreach(Persona persona in persone){
                persona.saluta1();
            }

            Insegnante insegnante1 = new Insegnante("Maria", "Esposito", 40, "Storia");
            Insegnante insegnante2 = new Insegnante("Elisa", "Verdi", 34, "Matematica");

            insegnante1.saluta1();
            insegnante2.saluta1();

        }
    }
}